### PR TITLE
Adjust docker image build

### PR DESCRIPTION
**Pull request by SuperQ**
* Add a Makefile.common step to expose docker image name.
* Use interpolated variable in CircleCI to setup docker image names.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
This will help @grobie's concerns about too much copy-paste in the new CircleCI template.

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,13 +32,15 @@ jobs:
...
-
     steps:
     - checkout
     - setup_remote_docker
+    - run:
+        name: Setup docker image names
+        command: |
+          echo 'export DOCKER_IMAGE_NAME="prom/$(make docker-image-name)"' >> $BASH_ENV
+          echo 'export QUAY_IMAGE_NAME="quay.io/prometheus/$(make docker-image-name)"' >> $BASH_ENV
+          source $BAH_ENV
```
[the remark this answers is not in the document]
Comment (SuperQ) [reply .circleci/config.yml]: lolz, *sigh*

**Comment by tomwilkie** [comment]
@SuperQ @bboreham found a nice way to reduce boilerplate in Cortex: https://github.com/weaveworks/cortex/blob/master/.circleci/config.yml#L3

**Comment by SuperQ** [comment]
@tomwilkie Nice, thanks, yaml templating is a great idea.  I'll add some of that.

**Comment by simonpasquier** [comment]
@SuperQ don't min if I close this PR as it has been addressed elsewhere?

**Pull request closed without merging**